syntax = "proto3";

package api.v1;

message Request {
    message EmptyRequest {}
    oneof request {
        EmptyRequest getInfo = 1;
        EmptyRequest getOtherThing = 2;
    }
}

message Response {
    oneof response {
        SysInfo info = 1;
        OtherThing other_thing = 2;
    }
}

message SysInfo {
  string serial_number = 1;
  string firmware_version = 2;
  string vendor = 3;
  string product = 4;
}

message OtherThing {
  string other = 1;
}